siriuspy/siriuspy/fofb/csdev.py

```diff
@@ -0,0 +1,298 @@
+
+class ETypes(_csdev.ETypes):
+    """Local enumerate types."""
+
+    OPEN_CLOSED = ('Open', 'Closed')
+
+    GLOB_INDIV = ('Global', 'Individual')
+
+    MEAS_RMAT_CMD = ('Start', 'Stop', 'Reset')
+    MEAS_RMAT_MON = ('Idle', 'Measuring', 'Completed', 'Aborted')
+
+    STS_LBLS_CORR = (
+        'Connected', 'PwrStateOn', 'OpModeConfigured', 'AccFreezeConfigured',
+        'InvRespMatRowSynced', 'LoopGainSynced')
+    STS_LBLS_FOFBCTRL = (
+        'Connected', 'BPMIdsConfigured', 'NetSynced', 'RefOrbSynced',
+        'TimeFrameLenSynced', 'BPMLogTrigsConfigured')
+
+
+_et = ETypes  # syntactic sugar
+
+
+# --- Const class ---
+
+class HLFOFBConst(_csdev.Const):
+    """Const class defining High Level FOFB constants."""
+
+    MIN_SING_VAL = 0.2
+    TIKHONOV_REG_CONST = 0
+    SINGVALHW_THRS = 1e-14
+
+    CONV_UM_2_NM = 1e3
+    ACCGAIN_RESO = 2**-12
+
+    LoopState = _csdev.Const.register('LoopState', _et.OPEN_CLOSED)
+    GlobIndiv = _csdev.Const.register('GlobIndiv', _et.GLOB_INDIV)
+    UseRF = _csdev.Const.register('UseRF', _et.DSBL_ENBL)
+    MeasRespMatCmd = _csdev.Const.register('MeasRespMatCmd', _et.MEAS_RMAT_CMD)
+    MeasRespMatMon = _csdev.Const.register('MeasRespMatMon', _et.MEAS_RMAT_MON)
+
+    def __init__(self):
+        """Class constructor."""
+
+        # device names and nicknames
+        self.bpm_names = _BPMSearch.get_names({'sec': 'SI', 'dev': 'BPM'})
+        self.ch_names = _PSSearch.get_psnames({'sec': 'SI', 'dev': 'FCH'})
+        self.cv_names = _PSSearch.get_psnames({'sec': 'SI', 'dev': 'FCV'})
+
+        self.bpm_nicknames = _BPMSearch.get_nicknames(self.bpm_names)
+        self.ch_nicknames = _PSSearch.get_psnicknames(self.ch_names)
+        self.cv_nicknames = _PSSearch.get_psnicknames(self.cv_names)
+
+        # device position along the ring
+        self.bpm_pos = _BPMSearch.get_positions(self.bpm_names)
+        self.ch_pos = _MASearch.get_mapositions(map(
+            lambda x: x.substitute(dis='MA'), self.ch_names))
+        self.cv_pos = _MASearch.get_mapositions(map(
+            lambda x: x.substitute(dis='MA'), self.cv_names))
+
+        # total number of devices
+        self.nr_bpms = len(self.bpm_names)
+        self.nr_ch = len(self.ch_names)
+        self.nr_cv = len(self.cv_names)
+        self.nr_chcv = self.nr_ch + self.nr_cv
+        self.nr_corrs = self.nr_chcv + 1  # include RF
+
+        # data folder
+        ioc_fol = _os.path.join(
+            '/home', 'sirius', 'iocs-log', 'si-ap-fofb', 'data')
+        self.reforb_fname = _os.path.join(ioc_fol, 'reforbit.orb')
+        self.respmat_fname = _os.path.join(ioc_fol, 'respmat.respmat')
+
+        # reforb and matrix parameters
+        self.reforb_size = self.nr_bpms
+        self.matrix_size = self.nr_corrs * (2 * self.nr_bpms)
+        self.nr_svals = min(self.nr_corrs, 2 * self.nr_bpms)
+        self.circum = 518.396  # in meter
+        self.harm_number = 864
+        self.rev_per = self.circum / 299792458  # in seconds
```

Comment:
I think we sould use `mathphys` for this constants

Reply:
We are not actually using these constants in this IOC, I will remove them

Reply:
done!